webapp/components/NewsList/NewsList.vue

```diff
@@ -62,16 +81,103 @@ export default {
     }
   },
   methods: {
-    updateitem (eventitem) {
-      this.items = this.items.map(item => eventitem.id === item.id ? eventitem : item)
+    async updateitem (eventitem) {
+      const upvotePost = gql`
+        mutation upvotePost($id: ID!) {
+          upvote(id: $id ) {
+            id
+            title
+            votes
+            author {
+              id
+            }
+          }
+        }
+      `
```

Comment:
I like to collect lengthy graphql mutations and queries in a file and import them in the required components.